src/test/java/dal/fatih/todoly/TodolyTest.java

```diff
@@ -1,252 +1,553 @@
 
-    public TodolyTest() throws SQLException {
-    }
+    @LocalServerPort
+    private int randomServerPort;
 
-    @Before
-    public void setUp() {
-        outContent = new ByteArrayOutputStream();
-        System.setOut(new PrintStream(outContent));
-    }
+    @Autowired
+    private TestRestTemplate testRestTemplate;
 
-    @After
-    public void tearDown() {
-        try {
-            Statement statement = connection.createStatement();
-            statement.executeUpdate("TRUNCATE TABLE task");
-            connection.close();
-            statement.close();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
+    @Autowired
+    private TaskRepository taskRepository;
 
     @Test
-    public void shouldNotAllowWrongSelection() {
-        provideInput(Arrays.asList("10", "q"));
-        App.main(new String[]{});
-        Assert.assertTrue(outContent.toString().contains("Invalid input"));
+    public void shouldCreateTask() throws URISyntaxException {
+        URI taskCreateUrl = new URI(createURLWithPort("/task"));
+        HttpEntity<TaskDTO> request = new HttpEntity<TaskDTO>(
+                new TaskDTO("Created-title-of-task", "Created-description-of-task"
+                        , LocalDateTime.now().plusDays(10))
+        );
+
+        ResponseEntity<String> responseEntity = this.testRestTemplate
+                .postForEntity(taskCreateUrl, request, String.class);
+        logger.warn(responseEntity.getBody());
+
+        String response = responseEntity.getBody();
+        assert response != null;
+        long createdTaskId = Long.parseLong(response.substring(6, response.length() - 1));
+
+        String actual = taskRepository.get(createdTaskId).getTitle();
+        String expected = "Created-title-of-task";
+
+        assertThat(actual, is(equalTo(expected)));
+        assertThat(responseEntity.getStatusCode(), equalTo(HttpStatus.CREATED));
     }
 
     @Test
-    public void shouldNotAllowEmptyTitle() {
-        provideInput(Arrays.asList("1", "", "description-of-task", "2030-05-05", "q"));
-        App.main(new String[]{});
-        Assert.assertTrue(outContent.toString().contains("Fill required fields"));
+    public void shouldAllowEmptyDescription() throws URISyntaxException {
+        URI taskCreateUrl = new URI(createURLWithPort("/task"));
+        HttpEntity<TaskDTO> request = new HttpEntity<TaskDTO>(
+                new TaskDTO("Title of task with empty description", null
+                        , LocalDateTime.now().plusDays(10))
+        );
+
+        ResponseEntity<String> responseEntity = this.testRestTemplate
+                .postForEntity(taskCreateUrl, request, String.class);
+        logger.warn(responseEntity.getBody());
+
+        String response = responseEntity.getBody();
+        assert response != null;
+        long createdTaskId = Long.parseLong(response.substring(6, response.length() - 1));

```

Comment:
Parse it as `HashMap` by using `ObjectMapper` from `fasterxml` since your API returns json. Extract it that way.